src/test/java/formflow/library/repository/UserFileRepositoryServiceTests.java

```diff
@@ -83,7 +73,12 @@ void shouldUserDocTypeSetInUserFileBuilderCall() {
         .docTypeLabel("BirthCertificate")
         .build();
 
-    UserFile savedFile = userFileRepositoryService.save(testFile);
+    UserFile savedFile = saveAndReload(testFile);
     assertThat(savedFile.getDocTypeLabel()).isEqualTo("BirthCertificate");
   }
+
+  private UserFile saveAndReload(UserFile testFile) {
+    UserFile savedFile = userFileRepositoryService.save(testFile);
```

Comment:
Save should have returned the updated object - either having updated the one handed in or a new object that's the result of an internal merge.  But you're not seeing the new object here?  But things function normally outside of tests?

Reply:
Yes, that is what I am seeing, contrary to expectation. It _could_ be something else we are doing wrong.

Reply:
You can see a similar pattern already in use here - https://github.com/codeforamerica/form-flow/blob/main/src/test/java/formflow/library/repository/SubmissionRepositoryServiceTest.java#L60-L62